### PR TITLE
4219 import location page related service hours

### PR DESCRIPTION
**Pull request by briaguya**
# Description

### What this does:
* When importing a location page, we also get service hours imported
### What this doesn't
* Actually connect the correct service page to the location page

https://github.com/cityofaustin/techstack/issues/4219

# Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How can this be tested?

 
see tests pass
try importing location pages with related services that have hours

# Checklist:
- [x] Request reviewers
- [ ] Slack-out message for review
- [ ] Link this PR in the issue
- [ ] Moved card to *review* in zenhub

### CODE REVIEW

File: joplin/pages/location_page/factories.py [hunk 1]
```diff
@@ -1,22 +1,53 @@
...
     def create(cls, *args, **kwargs):
-        # todo: put some stuff for related service hours here
         return super(LocationPageFactory, cls).create(*args, **kwargs)
 
+    @factory.post_generation
+    def add_related_services(self, create, extracted, **kwargs):
+        if extracted:
+            # A list of related services were passed in,
+            # this includes info about hours for the related service
+            # todo: actually link the related service
```
Comment (Niicck) [inline joplin/pages/location_page/factories.py:40]: What does actually linking mean?

File: joplin/importer/queries.py [hunk 2]
```diff
@@ -165,19 +165,19 @@ class GraphqlParser(Template):
...
-    #   edges {
-    #     node {
-    #       relatedService {
-    #         title
-    #       }
-    #       hoursSameAsLocation
-    #       $$$hours
-    #     }
-    #   }
-    # }
```
[the remark this answers is not in the document]
Comment (briaguya) [reply joplin/importer/queries.py:180]: it is (it's not commented out anymore)

File: joplin/pages/location_page/factories.py [hunk 3]
```diff
@@ -1,22 +1,53 @@
...
+            for location_page_related_service in extracted:
```
[the remark this answers is not in the document]
Comment (briaguya) [reply joplin/pages/location_page/factories.py:27]: same name in both places now

File: joplin/importer/queries.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Niicck) [reply joplin/importer/queries.py:180]: derp.

File: joplin/pages/location_page/factories.py [hunk 4]
```diff
@@ -73,9 +90,32 @@ def create_location_page_from_importer_dictionaries(page_dictionaries, revision_
...
+            related_service = ServicePage.objects.get(slug='placeholder_service_for_hours')
+        except ServicePage.DoesNotExist:
+            related_service = None
+        if not related_service:
+            related_service_dictionary = {
+                'parent': combined_dictionary['parent'],
+                'title': 'placeholder service for hours',
+                'slug': 'placeholder_service_for_hours'
+            }
+            related_service = ServicePageFactory.create(**related_service_dictionary)
```
Comment (Niicck) [inline joplin/pages/location_page/factories.py:114]: 👍

**Pull request merged**